// discord.utils/lib/structures/DiscordBotsList.js
const http = require("../utils/http.js")

module.exports = class DiscordBotsList {

	constructor(authorization) {
		this.authorization = authorization;
	}

	update(authorization) {
		this.authorization = authorization;
	}

	getBots() {
		return http.get("https://discordbots.org/api/bots", this.authorization);
	}

	getUser(user) {
		return http.get(`https://discordbots.org/api/users/${user}`, this.authorization)
	}

	getBot(bot, options) {
		if (!options) throw new Error("Must specify options for getBot method.")

		switch(options.method ? options.method.toLowerCase() : "get") {
			case "information":
				return http.get(`https://discordbots.org/api/bots/${bot}`, this.authorization)
				break;	
			case "statistics":
				return http.get(`https://discordbots.org/api/bots/${bot}/stats`, this.authorization)
				break;		
			case "post":
				if (!options.json) throw new Error("JSON Body keys not specified.");
				if (!options.authorization) throw new Error("Specific bot token not specified.")

				return http.post(`https://discordbots.org/api/bots/${bot}/stats`, options.authorization, {
					data : options.json
				})
				break;
			case "votes":
				return http.get(`https://discordbots.org/api/bots/${bot}/votes`, this.authorization)
				break;
			case "embed":
				return http.get(`https://discordbots.org/api/widget/${bot}.png`, this.authorization, {
					query : options
				})
				break;
			default:
				throw new Error("Method not found.")
		}
	}

	get auth() {
		return this.authorization;
	}

}
