// discord.utils/lib/structures/DiscordBots.js
const http = require("../utils/http.js")

module.exports = class DiscordBots {

	constructor(authorization) {
		this.authorization = authorization;
	}

	update(authorization) {
		this.authorization = authorization;
	}

	getBots() {
		return http.get("https://bots.discord.pw/api/bots", this.authorization);
	}

	getUser(user) {
		return http.get(`https://bots.discord.pw/api/users/${user}`, this.authorization)
	}

	getBot(bot, options) {
		if (!options) throw new Error("Must specify options for getBot method.")

		switch(options.method ? options.method.toLowerCase() : "get") {
			case "information":
				return http.get(`https://discordbots.org/api/bots/${bot}`, this.authorization)
				break;	
			case "statistics":
				return http.get(`https://bots.discord.pw/api/bots/${bot}/stats`, this.authorization)
				break;		
			case "post":
				if (!options.json) throw new Error("JSON Body keys not specified.");
				
				return http.post(`https://bots.discord.pw/api/bots/${bot}/stats`, this.authorization, {
					data : options.json
				})
				break;
			default:
				throw new Error("Method not found.")
		}
	}

	get auth() {
		return this.authorization;
	}

}
