// discord.utils/lib/structures/DiscordServices.js
const http = require("../utils/http.js")

module.exports = class DiscordServices {

	constructor(authorization) {
		this.authorization = authorization ? authorization : "dnZ4fFJbjtch56pNbfrZeSRfgWqdPDgf";
	}

	update(authorization) {
		this.authorization = authorization;
	}

	pat() {
		return http.get("https://nekos.life/api/pat", this.authorization, { authorization : "Key" })
	}

	hug() {
		return http.get("https://nekos.life/api/hug", this.authorization, { authorization : "Key" })
	}

	kiss() {
		return http.get("https://nekos.life/api/kiss", this.authorization, { authorization : "Key" })
	}

	neko() {
		return http.get("https://nekos.life/api/neko", this.authorization, { authorization : "Key" })
	}

	why() {
		return http.get("https://nekos.life/api/why", this.authorization, { authorization : "Key" })
	}

	lewd() {
		return http.get("https://nekos.life/api/lewd/neko", this.authorization, { authorization : "Key" })
	}

	lizard() {
		return http.get("https://nekos.life/api/lizard", this.authorization, { authorization : "Key" })
	}

	get auth() {
		return this.authorization;
	}

}
