// discord.utils/lib/structures/Terminal.js
const http = require("../utils/http.js")

module.exports = class Terminal {

	constructor(authorization) {
		this.authorization = authorization;
	}

	update(authorization) {
		this.authorization = authorization;
	}

	getBots() {
		return http.get("https://ls.terminal.ink/api/v1/bots", this.authorization);
	}

	getBot(bot, options) {
		if (!options) throw new Error("Must specify options for getBot method.")

		switch(options.method ? options.method.toLowerCase() : "get") {
			case "information":
				return http.get(`https://ls.terminal.ink/api/v1/bots/${bot}`, this.authorization)
				break;	
			case "post":
				if (!options.json) throw new Error("JSON Body keys not specified.");
				return http.post(`https://ls.terminal.ink/api/v1/bots/${bot}`, this.authorization, {
					data : options.json
				})
				break;
			case "embed":
				return http.get(`https://ls.terminal.ink/api/v1/bots/${bot}/embed`, {
					query : {
						width : options.width,
						locale : options.locale,
						type : options.type
					}
				})
				break;
			default:
				throw new Error("Method not found.")
		}
	}

	get auth() {
		return this.authorization;
	}

}
